test/breakfast_test.exs

```diff
@@ -120,6 +126,24 @@ defmodule BreakfastTest do
                }
              }
     end
+
+    test "should raise a runtime exception if the custom fetch returns a bad value", %{
+      params: params
+    } do
+      params = Map.put(params, :bad_return, true)
+
+      %Breakfast.FetchError{message: message, field: field, type: type, fetcher: fetcher} =
+        assert_raise Breakfast.FetchError, fn ->
+          Breakfast.decode(JSUser, params)
+        end
+
+      assert "Expected fetcher for `age` (`:fetch_age`) to return `{:ok, value}` or `:error` but got `:bad_return`" =
+               message
+
+      assert :age = field
+      assert :integer = type
+      assert :fetch_age = fetcher
```

Comment:
Do you write your assertions as matches in this way for a specific reason? I have no problem with it, just curious :)

Reply:
Not really, just an habit :stuck_out_tongue: I think the messages are not as helpful when there's an error with a pattern match vs a `==` with `assert` though so I should probably try to use `==` instead